### PR TITLE
Update shared credo config

### PR DESCRIPTION
**Pull request by Stratus3D**
[Card](https://rentpath.atlassian.net/browse/SRV-4853)

* Update version management instructions in Elixir readme
* Update credo config with new checks

I tested these updated checks on callrail-service and no warnings were generated.

### CODE REVIEW

File: elixir/.credo.exs [hunk 1]
```diff
@@ -79,51 +79,66 @@
         {Credo.Check.Readability.ModuleAttributeNames},
         {Credo.Check.Readability.ModuleDoc},
         {Credo.Check.Readability.ModuleNames},
-        {Credo.Check.Readability.StrictModuleLayout,
-          order: ~w/shortdoc moduledoc behaviour use import require alias module_attribute defstruct type typep callback macrocallback optional_callbacks/a},
+        {Credo.Check.Readability.NestedFunctionCalls},
```
Comment (Stratus3D) [inline elixir/.credo.exs:82]: This check was contributed by me a year ago. This enforces one of the rules from the Christopher Adams style guide.

File: elixir/.credo.exs [hunk 2]
```diff
@@ -79,51 +79,66 @@
...
         {Credo.Check.Readability.ParenthesesInCondition, false},
         {Credo.Check.Readability.ParenthesesOnZeroArityDefs},
         {Credo.Check.Readability.PredicateFunctionNames},
         {Credo.Check.Readability.PreferImplicitTry, exit_status: 0},
-        # {Credo.Check.Readability.PreferUnquotedAtoms}, # There is a bug in this Credo check in right now.
+        {Credo.Check.Readability.PreferUnquotedAtoms},
```
Comment (Stratus3D) [inline elixir/.credo.exs:87]: The bug has been fixed.

File: elixir/.credo.exs [hunk 3]
```diff
@@ -79,51 +79,66 @@
...
+        {Credo.Check.Readability.WithSingleClause},
 
         {Credo.Check.Refactor.ABCSize},
         {Credo.Check.Refactor.AppendSingleItem, exit_status: 0},
         {Credo.Check.Refactor.CaseTrivialMatches, false},
         {Credo.Check.Refactor.CondStatements, false},
         {Credo.Check.Refactor.CyclomaticComplexity},
         {Credo.Check.Refactor.DoubleBooleanNegation},
+        {Credo.Check.Refactor.FilterFilter},
+        {Credo.Check.Refactor.FilterReject},
```
Comment (Stratus3D) [inline elixir/.credo.exs:108]: Both of these checks warn about pipelines containing multiple calls when one is sufficient.

File: elixir/.credo.exs [hunk 4]
```diff
@@ -79,51 +79,66 @@
...
         {Credo.Check.Refactor.FunctionArity, max_arity: 6},
         {Credo.Check.Refactor.LongQuoteBlocks, false},
         {Credo.Check.Refactor.MapInto, false},
+        {Credo.Check.Refactor.MapJoin},
+        {Credo.Check.Refactor.MapMap},
```
Comment (Stratus3D) [inline elixir/.credo.exs:113]: Both of these checks warn about pipelines containing multiple calls when one is sufficient.

File: elixir/.credo.exs [hunk 5]
```diff
@@ -79,51 +79,66 @@
...
         {Credo.Check.Refactor.NegatedConditionsInUnless},
         {Credo.Check.Refactor.NegatedConditionsWithElse},
         {Credo.Check.Refactor.Nesting},
         {Credo.Check.Refactor.PerceivedComplexity, false},
         {Credo.Check.Refactor.PipeChainStart,
-         excluded_argument_types: [:atom, :binary, :fn, :keyword], excluded_functions: []},
+          excluded_argument_types: [:atom, :binary, :fn, :keyword], excluded_functions: []},
+        {Credo.Check.Refactor.RedundantWithClauseResult},
+        {Credo.Check.Refactor.RejectFilter},
+        {Credo.Check.Refactor.RejectReject},
```
Comment (Stratus3D) [inline elixir/.credo.exs:123]: Both of these checks warn about pipelines containing multiple calls when one is sufficient.

File: elixir/.credo.exs [hunk 6]
```diff
@@ -79,51 +79,66 @@
...
+        {Credo.Check.Refactor.MapMap},
         {Credo.Check.Refactor.MatchInCondition},
         {Credo.Check.Refactor.NegatedConditionsInUnless},
         {Credo.Check.Refactor.NegatedConditionsWithElse},
         {Credo.Check.Refactor.Nesting},
         {Credo.Check.Refactor.PerceivedComplexity, false},
         {Credo.Check.Refactor.PipeChainStart,
-         excluded_argument_types: [:atom, :binary, :fn, :keyword], excluded_functions: []},
+          excluded_argument_types: [:atom, :binary, :fn, :keyword], excluded_functions: []},
+        {Credo.Check.Refactor.RedundantWithClauseResult},
```
Comment (Stratus3D) [inline elixir/.credo.exs:121]: Warns about `with` statements with unnecessary `pattern <- expression` clauses in the head.

File: elixir/.credo.exs [hunk 7]
```diff
@@ -79,51 +79,66 @@
...
+        {Credo.Check.Refactor.RejectFilter},
+        {Credo.Check.Refactor.RejectReject},
         {Credo.Check.Refactor.VariableRebinding},
         {Credo.Check.Refactor.UnlessWithElse},
+        {Credo.Check.Refactor.WithClauses},
```
Comment (Stratus3D) [inline elixir/.credo.exs:126]: Warns about `with` clauses with head expressions that don't use the `<-` operator.

File: elixir/.credo.exs [hunk 8]
```diff
@@ -79,51 +79,66 @@
...
 
+        {Credo.Check.Warning.ApplicationConfigInModuleAttribute},
```
Comment (Stratus3D) [inline elixir/.credo.exs:128]: Warns about application config that would get compiled into modules. Understands the `compile_env` function.

File: elixir/.credo.exs [hunk 9]
```diff
@@ -79,51 +79,66 @@
...
         {Credo.Check.Warning.BoolOperationOnSameValues},
+        {Credo.Check.Warning.Dbg},
```
Comment (Stratus3D) [inline elixir/.credo.exs:130]: `Dbg` is something new, but like `IEx.Pry` or `IO.inspect` it should never be committed in the code.

File: elixir/.credo.exs [hunk 10]
```diff
@@ -79,51 +79,66 @@
...
         {Credo.Check.Warning.ExpensiveEmptyEnumCheck},
         {Credo.Check.Warning.IExPry},
         {Credo.Check.Warning.IoInspect},
         {Credo.Check.Warning.LazyLogging, false},
         {Credo.Check.Warning.MapGetUnsafePass},
+        {Credo.Check.Warning.MixEnv},
```
Comment (Stratus3D) [inline elixir/.credo.exs:136]: Warns about `Mix.env` being used in application code.

File: elixir/.credo.exs [hunk 11]
```diff
@@ -79,51 +79,66 @@
...
         {Credo.Check.Warning.OperationOnSameValues},
         {Credo.Check.Warning.OperationWithConstantResult},
         {Credo.Check.Warning.RaiseInsideRescue},
+        {Credo.Check.Warning.SpecWithStruct},
```
Comment (Stratus3D) [inline elixir/.credo.exs:140]: Expects only types to be present in typespecs. Will warn about struct literals in typespecs.

File: elixir/.credo.exs [hunk 12]
```diff
@@ -79,51 +79,66 @@
...
+        {Credo.Check.Warning.UnsafeToAtom},
```
Comment (Stratus3D) [inline elixir/.credo.exs:141]: Warns about any `to_atom` function. Requires us to switch to using `to_existing_atom` for safety.

**Review by pasha — APPROVED** [review]

**Review by pasha — APPROVED** [review]

**Review by KevinMarth — APPROVED** [review]

**Review by joshdayan — APPROVED** [review]

**Review by rkledesma — APPROVED** [review]

**Pull request merged**